app/helpers/application_helper.rb

```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+  def link_to_add_fields(name, f, association)
+    new_object = f.object.send(association).klass.new
+    id = new_object.object_id
+    fields = f.fields_for(association, new_object, child_index: id) do |builder|
+      render(association.to_s.singularize + "_fields", f: builder)
+    end
+    link_to(name, '#', class: "add_fields", data: {id: id, fields: fields.gsub("\n", "")})
```

Comment:
In "fields", are we saving all of the rendered fields? I'm not sure if I like that.
The usual way to manage this with JS is to render a template for the fields inside the form and duplicate that through JS:

```html
<form>
  <template>
    <div>
      <input /> <!-- template fields -->
    </div>
  </template>
  <div>
    <input /> <!-- real fields -->
  </div>
</form>
```

We could also attempt to add new fields by using Turbo, with a link to the child's new page, but I've already tried that, and it's not as straightforward as it might seem.

Reply:
can you check this [commit](https://github.com/AguirreCampiNicolas/onboarding/commit/5fb213f263e2dde5d8bb0fe02cfd06bd0e4d565b)? I ended up using [this stimulus component](https://www.stimulus-components.com/docs/stimulus-rails-nested-form/)  which was pretty easy to install.